### ISSUE TITLE
about the "orion_retro_low_latency" version

### ISSUE THREAD

**Issue by ahjolinna** [rakslice/orion#10]
thanks for updating the "retro" version instead....I like it more than the newer "material" version....thought this older version doesn't seem to have HIDPI support, also the .desktop file needs to be updated

**Comment by rakslice** [comment]
Yeah I mainly use this version too. I've been creating changes with the goal of also cherry picking them into this version. However eventually some breaking change may come along that I don't want to spend the time to update it for but for now it continues.

HIDPI: What is it that you want to be different specifically? For me the main place where it's an issue is the chat UI, and running on a ~275 dpi display (a 4k 13-inch laptop), enabling Qt's auto scaling (environment variable `QT_AUTO_SCREEN_SCALE_FACTOR=1`, see https://doc.qt.io/qt-5/highdpi.html) gives me reasonable sized images, and although the default text size with this is fine for me, the Chat Text Scale Factor selection in the options would allow me to make it larger if I needed to.

I don't use the stock .desktop file, but I can update it to match the other one.

**Comment by ahjolinna** [comment]
I use openSUSE -linux (I think you use windows?), when it comes to hidpi, the icons are just on the smaller size, but the font/chat & img are fine....nothing major but still

I don't know how you feel about this, but I think maybe [kirigami](https://www.kde.org/products/kirigami/) (from kde) could have been better UI solution than the current "material" one...at least if you want a mobile/tablet/touch version also....for example [Subsurface](https://subsurface-divelog.org/) uses it...and oc more KDE apps are moving to it...like [vvave](https://conf.kde.org/en/Akademy2018/public/events/13) music player

btw. I'm going to buy Sony's next flagship phone, the [Xperia XZ3](https://www.gsmarena.com/sony_xperia_xz3-review-1822.php), it has OLED, 2K, HDR, 10bit, 6.0-inch & 18:9 screen ...with latest (Sony TV) BARVIA tech. why? well I have been waiting for a Sony (or Nokia) phone with this kinda screen for a long time

**Comment by ahjolinna** [comment]
have you notice that there is a huge memory leak with this version, I just had over 7gb...or does it load the VODs/steams to ram?

**Comment by ahjolinna** [comment]
if you want I have Valgrind/callgrind output

**Comment by rakslice** [comment]
re memory leak: perhaps you didn't see the note https://github.com/rakslice/orion/blob/twitch_low_latency/src/network/prefetchstream.cpp#L17

**Comment by rakslice** [comment]
There may be other issues, I don't know. The m3u requesting is just happening from QML; I don't know if Qt's JS engine has any resource issues with an indefinite loop of XHR

**Comment by rakslice** [comment]
Also just in general: when you hit the pause button in Orion it just pauses the player with its socket still open. With this proxying for low latency streams, as long as the socket is still open, this proxy is still reading the stream and sending it down, and Qt does not limit write buffer size.

**Comment by ahjolinna** [comment]
Well that sucks, do you have any workaround ideas?

Also what do you think about KDE's kirigami project (doesn't require any kde dependencies), in short it's a qml add-on/extension

**Comment by rakslice** [comment]
Okay I've made a change to that, so that fragment fetching pauses when writes to the proxy socket are backing up (i.e. once mpv has read ahead whatever it is willing to read ahead and data is starting to accumulate on the Qt side, which is a few minutes for a typical source stream).  Polling the playlist to get new fragment URLs still happens, so that if/when the player is unpaused, it will eventually be able to play them (provided the fragment URLs are still valid when fetching reaches them).  That means fragment URLs will still be queued into memory while the player is paused, but this is several orders of magnitude less data than the fragment contents.

**Comment by rakslice** [comment]
My interest in Orion mainly came out of wanting something lighter weight and faster to use on the desktop than Twitch in a browser, at a time when their site and web player performed somewhat worse than it does now.  And just having it be separate from my browser audio stream and tab ecosystem is a plus.

This interest doesn't really translate to mobile/tablet devices, because they are effectively single-task-at-a-time devices, where there are no other things happening at the same time as watching a Twitch stream for it to impact, and anyway major platforms get a Twitch app that is already somewhat better than the browser-based Twitch.

**Comment by ahjolinna** [comment]
I will check the changes tomorrow when I wake up. 

And about kirigami, it is designed to be a dynamic UI solution between desktop and mobile, and as the creator orion for some reason wanted to make it happen for both platforms I personally think kirigami would have been a better solution than the current "material design" which favors mobile over desktop.

For me having a android/mobile version is fine, as long as it won't compromise the desktop experience (as it has now)... Because the official twitch app is kinda awful.

Also if orion used kirigami, it would scale better  between low dpi and hidpi displays, as it's designed to be more adaptive

I think there has been talk about adding/merging kirigami to Qt/qml

_Sent from my Xperia X Compact using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by ahjolinna** [comment]
sorry for the late response, It seems the "memory leak " fix/workaround works...good job
